torchvision/prototype/transforms/_geometry.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any, Dict, Tuple, Union
+
+import torch
+from torch.nn.functional import interpolate
+from torchvision.prototype.datasets.utils import SampleQuery
+from torchvision.prototype.features import BoundingBox, Image
+from torchvision.prototype.transforms import Transform
+
+
+class HorizontalFlip(Transform):
+    @staticmethod
+    def image(input: Image) -> Image:
+        return Image(input.flip((-1,)), like=input)
+
+    @staticmethod
+    def bounding_box(input: BoundingBox) -> BoundingBox:
+        x, y, w, h = input.convert("xywh").to_parts()
+        x = input.image_size[1] - (x + w)
+        return BoundingBox.from_parts(x, y, w, h, like=input, format="xywh")
```

Comment:
I'm not 100% sure about what `from_parts` is doing, but we should make sure that the format returned by `HorizontalFlip` is consistent with the format from `input` -- I assume `like` will take care of this here, and `format` will just take the format given to itself as `a, b, c, d` ?

Reply:
> I'm not 100% sure about what `from_parts` is doing

It is a convenience method, that creates a new `BoundingBox` from the individual parts. It is equivalent to do `BoundingBox(torch.stack((x, y, w, h), dim=-1), ...)`.

> but we should make sure that the format returned by `HorizontalFlip` is consistent with the format from `input`

It this a general requirement? My idea was to keep the format variable an each transform converts to the format that is most convenient for it. Given that the user will need to perform some "special" collation for bounding boxes anyway, IMO it is ok to ask them to convert it to whatever format they want to use then. For example

```python
dataset = datasets.load("foo").map(transform).map(my_special_collate(bbox_format="cxcywh"))
image_batch, bbox_batch = next(iter(dataset))
```

>  I assume `like` will take care of this here, and `format` will just take the format given to itself as `a, b, c, d` ?

Nope. By passing `format` explicitly, this takes precedence over `like`. To achieve what you want we need to add a `.convert(input.format)` after the creation. Note that we cannot drop the explicit `format`, because the data is in `xywh` format and we need create the `BoundingBox` with this format first before we can convert to any other format.

Reply:
> It this a general requirement?

This is generally a good thing to have, as it will avoid surprises to users. I would strongly recommend returning the same format as the input, even if internally we convert to a different format for performing the operation.

Imagine if you pass bounding boxes as `xyxy`, apply a `flip` operation, get a bounding box as `xywh`, and then cast it to Tensor to pass to the model? You won't even see that the bounding box was in `xywh` because it's implicit, and your model will be ingesting the wrong data.